kart/cli.py

```diff
@@ -318,6 +322,63 @@ def lfs(ctx, args):
     execvp("git", [*params, *args])
 
 
+@cli.command(
+    name="ext-run", context_settings=dict(ignore_unknown_options=True), hidden=True
+)
+@click.pass_context
+@click.argument("script", type=click.Path(exists=True, dir_okay=False), required=True)
+@click.argument("args", nargs=-1, type=click.UNPROCESSED)
+def ext_run(ctx, script, args):
+    """
+    Invoke the main() method from an arbitrary Python script in the Kart
+    environment. There is no public API for Kart.
+    """
+    script = pathlib.Path(script)
+    # strip all the suffixes
+    module_name = "kart.ext_run." + script.name.split(".", 1)[0]
+
+    def _print_ext_tb():
+        # print a traceback from the ext module down
+        extracts = traceback.extract_tb(sys.exc_info()[2])
+        count = len(extracts)
+        # find the first occurrence of the module file name
+        for i, extract in enumerate(extracts):
+            if extract[0] == script.name:
+                break
+            count -= 1
+        # keep only the count of last lines
+        traceback.print_exc(limit=-count)
+
+    spec = importlib.util.spec_from_file_location(module_name, script)
+    module = importlib.util.module_from_spec(spec)
+    try:
+        spec.loader.exec_module(module)
+    except Exception:
+        click.secho(f"Error loading {script}:", fg="red", file=sys.stderr)
+        _print_ext_tb()
+        raise click.Abort()
+
+    if not hasattr(module, "main") or not callable(module.main):
+        click.secho(
+            f"{script} does not have a main() function", fg="red", file=sys.stderr
+        )
+        raise click.Abort()
+
+    try:
+        f_sig = inspect.signature(module.main)
+        f_sig.bind(ctx=ctx, args=args)
+    except TypeError:
+        click.secho(
+            f"Invalid function definition for {script}:main()",
+            fg="red",
+            file=sys.stderr,
+        )
+        _print_ext_tb()
```

Comment:
Output is a bit weird here:

```
$ kart ext-run x.py
Invalid function definition for x.py:main()
Traceback (most recent call last):
TypeError: got an unexpected keyword argument 'args'
Aborted!
```
Maybe just show the full traceback? Or at least the ext_run frame itself